### PR TITLE
Convert language from using ID nums to ISO names

### PR DESCRIPTION
**Pull request by VSEphpbb**
phpBB seems to use the language's ISO name more than the language's ID. If we switch to using ISO code names we can more easily handle language stuff, such as referencing a board's default lang from the config array. This will also prevent problems if a user deletes, then reinstalls a language pack, which could result in orphaning the rules written in that language (because the ID will change, but the ISO code will remain the same).

### CODE REVIEW

File: entity/rule.php [hunk 1]
```diff
@@ -519,28 +519,45 @@ public function set_anchor($anchor)
...
+				$sql = 'SELECT lang_id
+					FROM ' . LANG_TABLE . "
+					WHERE lang_iso = '" . $this->db->sql_escape($language) . "'";
+				$result = $this->db->sql_query($sql);
+				$lang_id = $this->db->sql_fetchfield('lang_id');
+				$this->db->sql_freeresult($result);
+
+				if (!$lang_id)
+				{
+					throw new \phpbb\boardrules\exception\unexpected_value(array('rule_language', 'WRONG_DATA_LANG', 'UNEXPECTED_VALUE'));
```
Comment (rxu) [inline entity/rule.php]: ```WRONG_DATA_LANG``` appears unhandled if you set wrong language iso, like
rule_warn

File: migrations/v20x/m15_language_iso.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ * Board Rules extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2017 phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\boardrules\migrations\v20x;
+
+class m15_language_iso extends \phpbb\db\migration\container_aware_migration
```
Comment (rxu) [inline migrations/v20x/m15_language_iso.php]: Container seems not to be used in this migration 😛

File: entity/rule.php (hunk 1 above)
Comment (VSEphpbb) [inline entity/rule.php]: Yeah I know. I just don't care about that for now. The exceptions classes don't translate that.

File: migrations/v20x/m15_language_iso.php (hunk 2 above)
Comment (VSEphpbb) [inline migrations/v20x/m15_language_iso.php]: Belive me, there are new migration files inbound eventually once I figure a way around postgresql b.s. ;)

File: entity/rule.php (hunk 1 above)
Comment (VSEphpbb) [inline entity/rule.php]: OK, this has been solved too now.

**Pull request merged**